mono/metadata/icall.c

```diff
@@ -875,6 +875,20 @@ ves_icall_System_Array_CreateInstanceImpl (MonoReflectionTypeHandle type, MonoAr
 	return mono_array_new_full_handle (MONO_HANDLE_DOMAIN (type), aklass, sizes, lower_bounds, error);
 }
 
+#ifdef ENABLE_NETCORE
+gpointer
+ves_icall_System_Array_GetRawArrayGeometry(MonoArrayHandle arr, gint32* numComponents, gint32* elementSize, gint32* lowerBound, gint32* containsGCPointers, MonoError* error)
+{
+	MonoClass* arr_klass = mono_handle_class (arr);
+	MonoClass* elem_class = m_class_get_element_class (arr_klass);
+	*elementSize = mono_array_element_size (arr_klass);
+	*lowerBound = MONO_HANDLE_GETVAL (arr, bounds) ? MONO_HANDLE_GETVAL (arr, bounds [0].lower_bound) : 0;
+	*containsGCPointers = m_class_has_references (elem_class);
+	*numComponents = MONO_HANDLE_GETVAL (arr, max_length);
+	return (gpointer) MONO_HANDLE_RAW (arr)->vector;
+}
```

Comment:
This returns an interior pointer to a non-pinned object.

Reply:
@vargaz yes I was about to ask how to properly marshal it 🙁

Reply:
Do more/all the work in the icall instead?

Reply:
Or return it into an out handle?
Does it matter though? I mean, won't the reference get into managed soon enough and then work?
It is pinned at the point of return, right?

Reply:
Call Array.GetRawArrayData ().